traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -688,7 +688,7 @@ func (cg *TOCacheGroup) handleCoordinateUpdate() (*int, error, error, int) {
 	}
 
 	if err = cg.updateCoordinate(); err != nil {
-		return nil, nil, tc.DBError, http.StatusInternalServerError
+		return nil, err, tc.DBError, http.StatusInternalServerError
```

Comment:
Internal Server Errors must not be reported to the user. The bug as documented in Issue #4442 is that the Internal Server Error occurs when attempting to set a Cache Group's Name to the Name of a different, existing Cache Group. That should be a client error in the range [400,500) (400 seems the best fit) because the client can change something about their request to make it succeed. However, that's not the _only_ error that can be returned by `updateCoordinate`, so you can't just return whatever error that gives back.